prometheus_client/openmetrics/parser.py

```diff
@@ -537,8 +636,18 @@ def build_metric(name, documentation, typ, unit, samples):
             else:
                 raise ValueError("Invalid line: " + line)
         else:
-            sample = _parse_sample(line)
-            if sample.name not in allowed_names:
+            if typ == 'histogram':
+                # set to true to account for native histograms naming exceptions/sanitizing differences
+                is_nh = True
+                sample = _parse_nh_sample(line, tuple(type_suffixes['histogram']))
```

Comment:
Maybe for a future PR, but it feels a bit wrong to have to parse the entire sample as a native histogram, then bail and parse a normal sample. I wonder if instead we could have one function that parses everything consistently. I think it should be possible if we start on the left and work our way right parsing the name, then optional labels, then a value (either float or histogram - whitespace separates it from other fields), etc...

Reply:
I'd be grateful if we could leave this for another iteration, considering also the difficulties I've mentioned about parsing native histograms alone

Reply:
Yep, happy to have it in a future PR!